pkg/types/kubevirt/validation/platform.go

```diff
@@ -0,0 +1,64 @@
+package validation
+
+import (
+	"fmt"
+	"net"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/kubevirt"
+	"github.com/openshift/installer/pkg/validate"
+)
+
+// ValidatePlatform checks that the specified platform is valid.
+func ValidatePlatform(p *kubevirt.Platform, fldPath *field.Path, c *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if p.Namespace == "" {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("namespace"), p.Namespace, "namespace is required"))
+	}
+
+	if p.NetworkName == "" {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("networkName"), p.NetworkName, "networkName is required"))
+	}
+
+	if err := validate.IP(p.APIVIP); err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("APIVIP"), p.APIVIP, err.Error()))
+	} else if err := validateIPInMachineNetworkEntryList(c.MachineNetwork, p.APIVIP); err != nil {
+		allErrs = append(
+			allErrs, field.Invalid(
+				fldPath.Child("APIVIP"),
+				p.APIVIP,
+				fmt.Sprintf("validation of APIVIP in CIDR %s failed, with error: %v", c.MachineNetwork, err),
```

Comment:
I understand your point so I did the following:
1. `validateIPInMachineNetworkEntryList` doesn't validate invalid IP, because it validated before
2. The only error message that returned is `"IP must be in machine network range, machineNetworkEntryList: %s", machineNetworkEntryList`
3. At this point I use only the error message returned from `validateIPInMachineNetworkEntryList` err.Error()